src/main/java/org/apache/sling/servlethelpers/MockSlingHttpServletRequest.java

```diff
@@ -516,6 +529,36 @@ public RequestParameterMap getRequestParameterMap() {
         return params;
     }
 
+    public void addRequestParameter(String name, String value) {
+    	if (this.parameterMap.containsKey(name)) {
+    		List<MockRequestParameter> list = Arrays.asList(this.parameterMap.get(name));
```

Comment:
this code is redundant across all three methods of addRequestParameter - maybe put it in one helper method?

Reply:
Done.  This was a particularly good idea as one of the versions was correct but the other two had a bug in them.  Consolidating them made the bug obvious.  Thanks!